From b897a4ce1ee2169cb65d0e3d6c4d137dd1afb12f Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 22 Nov 2012 21:30:59 -0500
Subject: [PATCH] Remove Duplication Vulnerabilities#
 Remove CraftBukkit
 "Feature" of ItemStack's having infinite use if less than 0 stack size.

---
 src/main/java/net/minecraft/server/PlayerConnection.java      | 2 +-
 src/main/java/net/minecraft/server/PlayerInteractManager.java | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 7ca0acf..0aabc64 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -692,7 +692,7 @@ public class PlayerConnection extends Connection {
         }
 
         itemstack = this.player.inventory.getItemInHand();
-        if (itemstack != null && itemstack.count == 0) {
+        if (itemstack != null && itemstack.count <= 0) { // CraftBukkit - Handle infinite use item bug.
             this.player.inventory.items[this.player.inventory.itemInHandIndex] = null;
             itemstack = null;
         }
diff --git a/src/main/java/net/minecraft/server/PlayerInteractManager.java b/src/main/java/net/minecraft/server/PlayerInteractManager.java
index 55f9ffa..8dc94d4 100644
--- a/src/main/java/net/minecraft/server/PlayerInteractManager.java
+++ b/src/main/java/net/minecraft/server/PlayerInteractManager.java
@@ -360,7 +360,7 @@ public class PlayerInteractManager {
                 }
             }
 
-            if (itemstack1.count == 0) {
+            if (itemstack1.count <= 0) { // CraftBukkit - Handle infinite use item bug.
                 entityhuman.inventory.items[entityhuman.inventory.itemInHandIndex] = null;
             }
 
-- 
1.8.1.1

